Aptoide/aptoide-client-v8
# [feature/MOB-786] Highlighted Search App

**What does this PR do?**

   Implements highlighted search result in Search. What this means is that if a search result meets certain criteria (specified in the ticket), it should allow the user to download the app right away without going in the AppView.
  Because of this, the repository flow had to be changed, and also in this ticket a new presenter specifically for Downloads is introduced!
  While this new presenter may not yet fit every download scenario, please check if it meets the requirements for this highlighted purpose, for now. I also mean this in terms of analytics. 

**Database changed?**

   No

**How should this be manually tested?**

  Go to Search, and search for example "Gacha Life", or other that meet the criteria defined in the ticket. Then, download the app, and try to do other things meanwhile (like loading more, changing filters, going to the appview and coming back, opening screenshots etc.). Everything should work as expected, including the downloads control. Also test apps with AABs and OBBs (e.g. "Netflix" and "Creative Destruction", respectively).

**What are the relevant tickets?**

  [MOB-786](https://aptoide.atlassian.net/browse/MOB-786)

**Code Review Checklist**
